contract/contracts/Elections/Elections.sol

```diff
@@ -0,0 +1,309 @@
+    address indexed createdBy,
+    uint256 startDate,
+    uint256 endDate,
+    uint256 seatCount
+  );
+
+  event VoteSubmitted(uint256 indexed electionId, address indexed voter);
+
+  event ResultsPublished(uint256 indexed electionId, address[] winners);
+
+  // Custom errors
+  error ElectionNotFound();
+  error ElectionNotActive();
+  error ElectionEnded();
+  error AlreadyVoted();
+  error NotEligibleVoter();
+  error ResultsAlreadyPublished();
+  error ResultsNotReady();
+  error Unauthorized();
+
+  function initialize(address _owner) public initializer {
+    __Ownable_init(_owner);
+    __Pausable_init();
+    _nextElectionId = 1;
+  }
+
+  function pause() external onlyOwner {
+    _pause();
+  }
+
+  function unpause() external onlyOwner {
+    _unpause();
+  }
+
+  /**
+   * @dev Creates a new election
+   * @param title Election title
+   * @param description Election description
+   * @param startDate Election start timestamp
+   * @param endDate Election end timestamp
+   * @param seatCount Number of BOD seats (must be odd)
+   * @param candidates Array of candidates
+   * @param eligibleVoters Array of eligible voter addresses
+   * @return electionId The ID of the created election
+   */
+  function createElection(
+    string memory title,
+    string memory description,
+    uint256 startDate,
+    uint256 endDate,
+    uint256 seatCount,
+    address[] memory candidates,
+    address[] memory eligibleVoters
+  ) external onlyOwner whenNotPaused returns (uint256 electionId) {
+    ElectionUtils.validateSeatCount(seatCount);
+    ElectionUtils.validateDates(startDate, endDate);
+    ElectionUtils.validateCandidates(candidates, seatCount);
+    ElectionUtils.validateVoters(eligibleVoters);
+
+    electionId = _nextElectionId++;
+    ElectionTypes.Election storage election = _elections[electionId];
+
+    election.id = electionId;
+    election.title = title;
+    election.description = description;
+    election.createdBy = msg.sender;
+    election.startDate = startDate;
+    election.endDate = endDate;
+    election.seatCount = seatCount;
+    election.resultsPublished = false;
+    election.voteCount = 0;
+
+    for (uint256 i = 0; i < candidates.length; i++) {
+      election.isCandidate[candidates[i]] = true;
+      election.candidateList.push(candidates[i]);
+    }
+
+    for (uint256 i = 0; i < eligibleVoters.length; i++) {
+      election.isEligibleVoter[eligibleVoters[i]] = true;
+      election.voterList.push(eligibleVoters[i]);
+    }
+
+    _electionIds.push(electionId);
+
+    emit ElectionCreated(electionId, title, msg.sender, startDate, endDate, seatCount);
+
+    return electionId;
+  }
+
+  /**
+   * @dev Submits a rank-order vote for an election
+   * @param electionId The election ID
+   * @param rankedCandidates Array of candidate addresses in order of preference
+   */
+  function submitRankOrderVote(
+    uint256 electionId,
+    address[] memory rankedCandidates
+  ) external whenNotPaused {
```

Comment:
A Member Vote for on candidate. 
At the end of the election we have an order of candidate. 

Ex: Canditate 1: 200 Votes
Candidate 2: 150 Votes
Candidate 3: 20 Votes

Reply:
A member can not vote, on an upcoming election
A Member Can not vote on a Past Election